### PR TITLE
Replace vosk-lib with vosk-koffi

### PR DESCRIPTION
**Pull request by tocha688**
vosk-koffi is based on koffi while vosk-lib is based on ffi-napi, and it is well known that ffi-napi has a lot of problems.

### CODE REVIEW

**Comment by tocha688** [comment]
vosk-koffi does not require the python environment to be installed and other additional environments

**Comment by JacobLinCool** [comment]
Hello @tocha688,

You are right; ffi-napi is problematic, and node-gyp is frustrating.

I wasn't aware of koffi before, but it appears to offer much better performance and usability than ffi-napi, which is excellent!

I just looked into your vosk-koffi, and it seems that the package size is quite large compared to vosk-lib because it bundles the shared libraries for three platforms into one package. I think using a post-install script to download platform-dependent libraries from the official release could significantly reduce the package size. Additionally, I would prefer not to include and use binaries from an unknown source, which is a terrible idea. Please use the officially released ones or have trackable ways to build them (such as using GitHub Actions...).

After these problems are solved, we can definitely replace vosk-lib with vosk-koffi.

Also, this package uses pnpm as its package manager during development. Please remove the yarn.lock file and update the pnpm-lock.yaml file.

Thanks.

File: src/solve.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ function reconize(dir: string): Promise<string> {
                 if (end_of_speech) {
                     const result = rec
                         .result()
-                        .alternatives.sort((a, b) => b.confidence - a.confidence)[0].text;
+                        .alternatives.sort((a:any, b:any) => b.confidence - a.confidence)[0].text;
```
Comment (k725) [inline src/solve.ts:216]:
```suggestion
                        .alternatives.sort((a, b) => b.confidence - a.confidence)[0].text;
```

`a` and `b` type is `RecognitionResults | (RecognitionResults & SpeakerResults)`.
I think not need this `any` cast.

**Comment by k725** [comment]
@JacobLinCool 

`node-ffi-napi`, which `vosk-lib` depends on, is completely broken in the latest version of Node 22. [^1]
Personally, I think this PR proposal is good. (Currently, I think it is unavoidable that the bundle size will increase.)
I'm also looking forward to the new version being published to npm.

[^1]: https://github.com/node-ffi-napi/node-ffi-napi/issues/269

**Comment by JacobLinCool** [comment]
@k725

It seems that @tocha688 has done something to reduce the package size.

I am going to merge this.

**Comment by JacobLinCool** [comment]
Released with v0.3.0

**Pull request merged**